activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -401,11 +401,17 @@ def type_cast_calculated_value(value, type, operation = nil)
         case operation
         when "count"   then value.to_i
         when "sum"     then type.deserialize(value || 0)
-        when "average" then value.respond_to?(:to_d) ? value.to_d : value
+        when "average" then cast_average(value)
```

Comment:
Just enough `value&.respond_to?(:to_d) ? value.to_d : value` or `value && value.respond_to?(:to_d) ? value.to_d : value`.

```suggestion
        when "average" then value && value.respond_to?(:to_d) ? value.to_d : value
```

Reply:
Thanks for the suggestion @kamipo change applied. 🙏